src/mongocxx/private/client_encryption.hh

```diff
@@ -142,6 +144,200 @@ class client_encryption::impl {
+            bson_master_key ? bson_master_key->bson() : nullptr,
+            result_ptr.get(),
+            &error);
+
+        if (!r) {
+            throw_exception<operation_exception>(error);
+        }
+
+        const bson_t* bulk_write_result =
+            libmongoc::client_encryption_rewrap_many_datakey_result_get_bulk_write_result(
+                result_ptr.get());
+
+        if (bulk_write_result) {
+            const auto doc =
+                bsoncxx::document::view(bson_get_data(bulk_write_result), bulk_write_result->len);
+            const auto val = bsoncxx::document::value(doc);
+            result::bulk_write bulk_write_result{val};
+            return result::rewrap_many_datakey(bulk_write_result);
+        } else {
+            return result::rewrap_many_datakey();
+        }
+    }
+
+    result::delete_result delete_key(bsoncxx::types::bson_value::view_or_value id) {
+        using bsoncxx::builder::basic::kvp;
+        using bsoncxx::builder::basic::make_document;
+
+        bson_error_t error;
+        libbson::scoped_bson_t reply_ptr;
+
+        bson_value_t libbson_key;
+
+        convert_to_libbson(&libbson_key, id);
+
+        const auto r = libmongoc::client_encryption_delete_key(
+            _client_encryption_t, &libbson_key, reply_ptr.bson_for_init(), &error);
+
+        bson_value_destroy(&libbson_key);
+
+        if (!r) {
+            throw_exception<operation_exception>(error);
+        }
+
+        const bsoncxx::document::value val = reply_ptr.steal();
+
+        // The C driver calls this field "deletedCount", but the C++ driver
+        // refers to this as "nRemoved". Make a new document with the field name
+        // changed to get around this.
+        //
+        // See: mongo-cxx-driver/src/mongocxx/result/bulk_write.cpp
+        // Function: std::int32_t bulk_write::deleted_count() const {
+        //     return view()["nRemoved"].get_int32();
+        // }
+        const auto new_val = make_document(kvp("nRemoved", val["deletedCount"].get_int32()));
+        return result::delete_result(result::bulk_write(new_val));
+    }
+
+    stdx::optional<bsoncxx::document::view_or_value> get_key(
+        bsoncxx::types::bson_value::view_or_value id) {
+        bson_error_t error;
+        libbson::scoped_bson_t key_doc;
+
+        bson_value_t libbson_value;
+
+        convert_to_libbson(&libbson_value, id);
+
+        const auto r = libmongoc::client_encryption_get_key(
+            _client_encryption_t, &libbson_value, key_doc.bson_for_init(), &error);
+
+        const auto cleanup = [&]() { bson_value_destroy(&libbson_value); };
+
+        if (!r) {
+            cleanup();
+            throw_exception<operation_exception>(error);
+        }
+
+        const bsoncxx::document::view_or_value val(key_doc.steal());
+
+        cleanup();
+        return val.view().empty() ? stdx::nullopt
+                                  : stdx::optional<bsoncxx::document::view_or_value>{val};
```

Comment:
```cpp
cleanup();

return key_doc.view().empty()
            ? stdx::nullopt
            : stdx::optional<bsoncxx::document::value>(key_doc.steal());
```
Avoid redundant copies in functions that should return `stdx::optional<bsoncxx::document::value>` in this file.